From d3300c29ec267e6227c5531843c95d092f5f9494 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 11:40:44 -0400
Subject: [net] ethtool: constify array pointer parameters to ethtool_ops::set_rxfh

Message-id: <1410176446-10440-9-git-send-email-ivecera@redhat.com>
Patchwork-id: 91654
O-Subject: [RHEL7.1 PATCH 08/10] ethtool, be2net: constify array pointer parameters to ethtool_ops::set_rxfh
Bugzilla: 1138823
RH-Acked-by: Michal Schmidt <mschmidt@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: #1138823

Backport note:
The be2net part of the patch is included in be2net driver update.

Upstream commit(s):
commit 33cb0fa7888510b5bd2096352b200cfe29db10fe
Author: Ben Hutchings <ben@decadent.org.uk>
Date:   Thu May 15 02:01:23 2014 +0100

    ethtool, be2net: constify array pointer parameters to ethtool_ops::set_rxfh

    Signed-off-by: Ben Hutchings <ben@decadent.org.uk>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/ethtool.h b/include/linux/ethtool.h
index 56adf9c..6dbccba 100644
--- a/include/linux/ethtool.h
+++ b/include/linux/ethtool.h
@@ -271,7 +271,8 @@ struct ethtool_ops {
 #else
  u32 (*get_rxfh_key_size)(struct net_device *);
  int (*get_rxfh)(struct net_device *, u32 *indir, u8 *key);
- int (*set_rxfh)(struct net_device *, u32 *indir, u8 *key);
+ int (*set_rxfh)(struct net_device *, const u32 *indir,
+       const u8 *key);
 #endif
  void   (*rh_reserved4)(void);
  void   (*rh_reserved5)(void);
-- 
1.7.1